### ISSUE TITLE
cover images returning null inconsistently from Docs

### ISSUE THREAD

**Issue by linoxyn** [cedricdelpoux/gatsby-source-google-docs#56]
**The npmPackages:**
```
gatsby: ^2.20.14 => 2.20.14
    gatsby-cli: ^2.11.5 => 2.11.5
    gatsby-image: ^2.3.2 => 2.3.2
    gatsby-plugin-catch-links: ^2.2.1 => 2.2.1
    gatsby-plugin-manifest: ^2.3.3 => 2.3.3
    gatsby-plugin-offline: ^3.1.2 => 3.1.2
    gatsby-plugin-prefetch-google-fonts: ^1.4.3 => 1.4.3
    gatsby-plugin-react-helmet: ^3.2.1 => 3.2.1
    gatsby-plugin-sharp: ^2.5.4 => 2.5.4
    gatsby-plugin-styled-components: ^3.2.1 => 3.2.1
    gatsby-plugin-typography: ^2.4.1 => 2.4.1
    gatsby-remark-embed-video: ^2.0.1 => 2.0.1
    gatsby-remark-images: ^3.2.2 => 3.2.2
    gatsby-remark-responsive-iframe: ^2.3.1 => 2.3.1
    gatsby-source-filesystem: ^2.2.2 => 2.2.2
    gatsby-source-google-docs: ^2.0.0-beta.6 => 2.0.0-beta.6
    gatsby-transformer-remark: ^2.7.1 => 2.7.1
    gatsby-transformer-sharp: ^2.4.4 => 2.4.4
```
When `gatsby develop` is run the Docs in a folder, each have a header, 'Different first page' is checked, all have an image, in a query not all images are returned:

```
"cover": {
                "image": null
              }
```
If I edit one of the Docs that has returned `null` for the image and then run `develop` that Doc may now return the data for the image, though now other Docs that returned an image before may now show in the query as `null`. I cannot see that there is any pattern to this behaviour. Only that a change to what returns either, the image data, or a null, changes only if there's been an edit or an addition of a new Doc. Running `gatsby develop` builds the same number of complete images each time it's run.  I have also run `gatsby clean` - this does not alter this behavior. 

**Example in GraphiQL:** for the [public Google folder](https://drive.google.com/open?id=1DAXgLypP-GMIAeVCVvTwKKHDOuqg75HYEPC9imHgOvA) 
```
query MyQuery {
  allMarkdownRemark(filter: {frontmatter: {page: {eq: "news"}}}, sort: {fields: frontmatter___year, order: DESC}, limit: 10) {
    edges {
      node {
        frontmatter {
          name
          cover {
            image {
              childImageSharp {
                fluid(maxWidth: 300) {
                  src
                }
              }
            }
          }
        }
      }
    }
  }
}
```
**Returns: (these should all return the image)**
```
{
  "data": {
    "allMarkdownRemark": {
      "edges": [
        {
          "node": {
            "frontmatter": {
              "name": "Tony Luciani's \"The Dressmaker\" awarded 1st prize in the 10th International \"FIGURATIVAS 2019\" Painting & Sculpture Competition",
              "cover": {
                "image": null
              }
            }
          }
        },
        {
          "node": {
            "frontmatter": {
              "name": "Dr. Roberta Bondar Celebrates 27th Anniversary of Space Launch",
              "cover": {
                "image": {
                  "childImageSharp": {
                    "fluid": {
                      "src": "/static/96369c2d10bd5db831e4e649bbe18067/35801/google-docs-image-0d6acd37-3643-5107-9f42-0d6a82070b96.jpg"
                    }
                  }
                }
              }
            }
          }
        },
        {
          "node": {
            "frontmatter": {
              "name": "Tony Luciani \"Inside My Head\" on CBC Radio",
              "cover": {
                "image": null
              }
            }
          }
        },
        {
          "node": {
            "frontmatter": {
              "name": "W. David Ward \"Eyes of Society: Art, Traditional Knowledge, and the Watchmen of Haida Gwaii\"",
              "cover": {
                "image": null
              }
            }
          }
        },
        {
          "node": {
            "frontmatter": {
              "name": "Shannon Craig Morphew: Featured in the Dulwich Picture Gallery's \"In View\" magazine",
              "cover": {
                "image": {
                  "childImageSharp": {
                    "fluid": {
                      "src": "/static/c49446a06709af5cb7f8690051fcdba3/9dc27/google-docs-image-84b6ba87-eb38-5a04-a9f1-b87710471aaf.jpg"
                    }
                  }
                }
              }
            }
          }
        },
        {
          "node": {
            "frontmatter": {
              "name": "The Art of the Deal",
              "cover": {
                "image": {
                  "childImageSharp": {
                    "fluid": {
                      "src": "/static/c48a9a2c19722a9a43e32c0db0569c84/f836f/google-docs-image-5f658c01-c856-564c-8019-637d9cec7a81.jpg"
                    }
                  }
                }
              }
            }
          }
        },
        {
          "node": {
            "frontmatter": {
              "name": "The Art of Salesmanship",
              "cover": {
                "image": {
                  "childImageSharp": {
                    "fluid": {
                      "src": "/static/383aa78c3dc0f0577654850811045a7b/f836f/google-docs-image-6998840a-0d40-5e3c-9959-6b2d10ab6d90.jpg"
                    }
                  }
                }
              }
            }
          }
        },
        {
          "node": {
            "frontmatter": {
              "name": "A rare painting by Paul Kane sets a new record for Canadian Art",
              "cover": {
                "image": null
              }
            }
          }
        }
      ]
    }
  }
}
```
Thank you

**Comment by linoxyn** [comment]
I was looking at some of the code to try to understand what is going on. I found this happens periodically.

In the `converters.js` file I am logging the **Document ID** and the **Image**.
```
function getCover(document) {
  const {
    headers,
    documentStyle,
    documentId
  } = document;
  console.log('DOC ID:', documentId);
  if (!documentStyle || !documentStyle.firstPageHeaderId || !headers[documentStyle.firstPageHeaderId]) {
      return null;
    }
    const headerElement = _get(headers[documentStyle.firstPageHeaderId], ["content", 0, "paragraph", "elements", 0]);
  
  const image = getImage(document, headerElement);
  console.log('IMAGE:', image);
  return image ? {
    image: image.source,
    title: image.title,
    alt: image.alt
  } : null;
}
```
In the console it outputs this for one of the queried documents, that has a header/cover/image, that is failing with **null**

```
DOC ID: 1xDGR6EJFVO2Qnr8vfMpLe_M5ltQszyk8rQNdS3yDKxc
IMAGE: {
  source: 'https://lh5.googleusercontent.com/tL4Sgk_HhHxd7sgD1zMMYZ-HOheTjk2PNLzFueQJXC-tnC0oeozeBvzBojPkwidYO9pLwOlf6hhbaDcr-tqYDoDi-DwlnOpDO0_VSyFV0HFg20uMbdtz2ifV-FCLNNqLl3K0I-wn0dyFpin4AQ',
  title: '',
  alt: ''
}
```
At the time I'm posting this the link above gives a browser **"403 error." "Your client does not have permission to get URL."**

Though if I use [Google's API request](https://developers.google.com/docs/api/reference/rest/v1/documents/get?apix_params=%7B%22documentId%22%3A%221xDGR6EJFVO2Qnr8vfMpLe_M5ltQszyk8rQNdS3yDKxc%22%2C%22suggestionsViewMode%22%3A%22DEFAULT_FOR_CURRENT_ACCESS%22%7D) for the same Doc, the image properties for the header/cover/image is not the same. Though here both the `contentUri` and `sourceUri`  links open the proper image without any errors.   

```
"imageProperties": {
            "contentUri": "https://lh3.googleusercontent.com/KX-KRDkJoxCA1Kr2d-XP2Uil_AmiN7i6a-YoB5WyARqAX0DVMa8aDe4oS5zI7TFWyFDWPAQISlsE6a-6R7eF1xcVfvBUghgq06wxqSKGn1SoM7QYOUByzWQs7D4GsplHkoIGexSNL3IqVegCLg",
            "sourceUri": "https://lh3.google.com/u/0/d/1dPIKRGvikFMKuttSZ8FK6Eoezrst9z27=s2048",
            "documentId": "1xDGR6EJFVO2Qnr8vfMpLe_M5ltQszyk8rQNdS3yDKxc"
```

I have tried to find where the wrong image source content is originating from though I'm not having much luck tracing this. The errors inside the app `Cannot read property 'childImageSharp' of null` are randomly generated after adding or editing even just one Doc.

I hope this helps.

thank you

**Comment by cedricdelpoux** [comment]
Could you try to generate a new token. with full permission?

**Comment by linoxyn** [comment]
@cedricdelpoux , I am unsure as to how I can give the new token "full permission"? I did create a quick simple [repo](https://github.com/linoxyn/test-google-docs) with a new project and with a new Token. I made a folder with 4 test Docs to query `cover` and the same random behaviour where the images from the header return null or not occurs.

Thank you.

**Comment by nascode** [comment]
i got this bug too.

node.frontmatter.cover.image only work sometime

For now i use node.document.cover.image, which is without image sharp but works everytime

**Comment by cedricdelpoux** [comment]
Than you for your help @nascode

**Comment by cedricdelpoux** [comment]
Fixed in `2.0.0-beta.9`
